pkg/client/observatorium/observability_config.go

```diff
@@ -80,6 +70,73 @@ type ObservabilityCloudWatchLoggingConfig struct {
 	configFilePath                string                                                      `validate:"-"`
 }
 
+type DataPlaneObservabilityOIDCCredentials struct {
+	ClientID     string `yaml:"client_id" validate:"required"`
+	ClientSecret string `yaml:"client_secret" validate:"required"`
+}
+
+type DataPlaneObservabilityOIDCStatic struct {
+	AuthorizationServer string                                `yaml:"authorization_server" validate:"required"`
+	Realm               string                                `yaml:"realm" validate:"required"`
+	Credentials         DataPlaneObservabilityOIDCCredentials `yaml:"credentials" validate:"dive"`
+}
+
+type DataPlaneObservabilityOIDCAutogenerated struct{}
+
+type DataPlaneObservabilityRemoteWriteAuthTypeOIDC struct {
+	Autogenerated *DataPlaneObservabilityOIDCAutogenerated `yaml:"autogenerated_configuration" validate:"omitempty"`
+	Static        *DataPlaneObservabilityOIDCStatic        `yaml:"static_configuration" validate:"omitempty,dive"`
+}
+
+type DataPlaneObservabilityRemoteWriteAuthentication struct {
+	OIDC *DataPlaneObservabilityRemoteWriteAuthTypeOIDC `yaml:"oidc" validate:"omitempty,dive"`
+}
+
+type DataPlaneObservabilityRemoteWriteConfiguration struct {
+	RemoteWriteURL string                                          `yaml:"remote_write_url" validate:"required"`
+	Authentication DataPlaneObservabilityRemoteWriteAuthentication `yaml:"authentication" validate:"required,dive"`
+}
+
+type DataPlaneObservabilityConfig struct {
+	configFilePath string
+	Enabled        bool
+
+	RemoteWriteConfiguration DataPlaneObservabilityRemoteWriteConfiguration `yaml:"remote_write_configuration" validate:"dive"`
+	GithubResourcesAuthToken string
+}
+
+func (c *DataPlaneObservabilityConfig) HasAutogeneratedOIDCConfiguration() bool {
+	if c.RemoteWriteConfiguration.Authentication.OIDC != nil {
+		return c.RemoteWriteConfiguration.Authentication.OIDC.Autogenerated != nil
+	}
+
+	return false
+}
+
+func (c *DataPlaneObservabilityConfig) HasStaticOIDCConfiguration() bool {
+	if c.RemoteWriteConfiguration.Authentication.OIDC != nil {
+		return c.RemoteWriteConfiguration.Authentication.OIDC.Static != nil
+	}
+
+	return false
+}
+
+func (c *DataPlaneObservabilityRemoteWriteAuthTypeOIDC) GetStaticRealm() string {
```

Comment:
Why are this getters being configured?

Reply:
I can remove them. I think initially I had very long names for the struct fields, but using the fields directly instead of the getters is almost identical now.